mapadroid/worker/MITMBase.py

```diff
@@ -55,123 +56,83 @@ def __init__(self, args, dev_id, origin, last_known_state, communicator: Abstrac
                                                  99)
         self._enhanced_mode = self.get_devicesettings_value('enhanced_mode_quest', False)
 
-    def _check_data_distance(self, data):
-        max_radius = self._mapping_manager.routemanager_get_max_radius(self._routemanager_name)
-        if not max_radius:
-            return True
-        max_radius: int = self._applicationArgs.maximum_valid_distance \
-            if max_radius < self._applicationArgs.maximum_valid_distance else max_radius
-        mode = self._mapping_manager.routemanager_get_mode(self._routemanager_name)
-        if mode in ["mon_mitm", "iv_mitm"]:
-            data_to_check = "wild_pokemon"
-        else:
-            data_to_check = "forts"
-        lat_sum, lng_sum, counter = 0, 0, 0
-
-        if data_to_check == "forts":
-            for cell in data:
-                if cell[data_to_check]:
-                    cell_id = cell["id"]
-                    if cell_id < 0:
-                        cell_id = cell_id + 2 ** 64
-                    lat, lng, _ = S2Helper.get_position_from_cell(cell_id)
-                    counter += 1
-                    lat_sum += lat
-                    lng_sum += lng
-        else:
-            for cell in data:
-                for element in cell[data_to_check]:
-                    counter += 1
-                    lat_sum += element["latitude"]
-                    lng_sum += element["longitude"]
-
-        if counter == 0:
-            return None
-        avg_lat = lat_sum / counter
-        avg_lng = lng_sum / counter
-        distance = get_distance_of_two_points_in_meters(float(avg_lat),
-                                                        float(avg_lng),
-                                                        float(self.current_location.lat),
-                                                        float(self.current_location.lng))
-        if distance > max_radius:
-            self.logger.debug2("Data is too far away!! avg location {}, {} from  data with self.current_location "
-                               "location {}, {} - that's a {}m distance with max_radius {} for mode {}", avg_lat,
-                               avg_lng, self.current_location.lat, self.current_location.lng, distance, max_radius,
-                               mode)
-            return False
-        else:
-            self.logger.debug("Data distance is ok! found avg location {}, {} from data with self.current_location "
-                              "location {}, {} - that's a {}m distance with max_radius {} for mode {}", avg_lat,
-                              avg_lng, self.current_location.lat, self.current_location.lng, distance, max_radius, mode)
-            return True
-
-    def _wait_for_data(self, timestamp: float = None, proto_to_wait_for=106, timeout=None):
+    def _wait_for_data(self, timestamp: float = None, proto_to_wait_for=106, timeout=None) \
```

Comment:
This function is doing a ton of work. We should split it out into pure functions to allow for easy unit-testing to confirm we have the correct results